src/build_vehicle_gui.cpp

```diff
@@ -869,7 +869,7 @@ static std::optional<std::string> GetNewGRFAdditionalText(EngineID engine)
 	}
 
 	StartTextRefStackUsage(grffile, 6);
-	std::string result = GetString(GetGRFStringID(grffile->grfid, 0xD000 + callback));
+	std::string result = GetString(GetGRFStringID(grffile->grfid, GRFStringID{0xD000 + callback}));
```

Comment:
I know... scope creep, but would it make sense to let `0xD000` be a named constant of type `GRFStringID`? Then the addition ought to make the `GRFStringID` without the constructor or narrowing warning, I think.